src/landscape.cpp

```diff
@@ -1060,7 +1060,7 @@ static bool RiverMakeWider(TileIndex tile, void *data)
 	/* If the tile is at height 0 after terraforming but the ocean hasn't flooded yet, don't build river. */
 	if (GetTileMaxZ(tile) == 0) return false;
 
-	TileIndex origin_tile = *(TileIndex *)data;
+	TileIndex origin_tile = *(TileIndex *)user_data;
```

Comment:
As you are touching this line, please switch it to the static_cast version.

Reply:
Mind if I do that in a separate PR. There are other places in the river gen code with the old cast.